media/src/App.svelte

```diff
@@ -576,6 +666,108 @@
         socket = makeSocket(roomId, handleSocketMessage);
     }
 
+    const mouseVector = new THREE.Vector2();
+    let selectedMesh;
+    let tangentParSurf;
+    let tangentLevel;
+    let tangentFuncGraph;
+
+    const onMouseMove = function (e) {
+        if (selectedObject) {
+            // normalized mouse coordinates
+            mouseVector.x = 2 * (e.clientX / window.innerWidth) - 1;
+            mouseVector.y = 1 - 2 * (e.clientY / window.innerHeight);
+
+            raycaster.setFromCamera(mouseVector, camera);
+            selectedMesh = scene.getObjectByName(selectedObject).parent;
+
+            const intersects = raycaster.intersectObjects(
+                [selectedMesh.children[0], selectedMesh.children[1]],
+                true
+            );
+
+            if (intersects.length > 0) {
+                const intersect = intersects[0];
+                point.position.x = intersect.point.x; 
+                point.position.y = intersect.point.y;
+                point.position.z = intersect.point.z;
+
+                const uv = intersect.uv;
+
+                switch (selectedMesh.type){
+                    case 'ParametricSurface':
+                        tangentParSurf({uv});
+                        break;
+                    case 'LevelSurface':
+                        tangentLevel({point})
+                        break;
+                    case 'FunctionGraph':
+                        tangentFuncGraph();
+                        break;
+                }
+
+                render();
+            }
+        }
+    };
+
+    const shiftDown = (e) => {
+        if (shadeUp) {
+            switch (e.key) {
+                case 'Backspace':
+                    selectedMesh.visible = !selectedMesh.visible;
+                    render();
+                    break;
+                case 'Shift':
+                    window.addEventListener('mousemove', onMouseMove, false);
```

Comment:
We already have an onPointerMove listener happening in App.js. `pointermove` is a more generic version of `mousemove`, and we shouldn't have both going on here. We'll have to think more about how we want this to work.

Reply:
If you want to combine the `pointermove` handles object selection and is always active, `mousemove` sets the point on the object surface and is only active when using shift. If you want me to merge the two do you still want the `mousemove` fuction, what I will call `setPoint()`, to only modify the point when shift is held?